pvanalytics/quality/time.py

```diff
@@ -184,3 +184,63 @@ def shifts_ruptures(event_times, reference_times, period_min=2,
     # localize the shift_amount series to the timezone of the input
     shift_amount = shift_amount.tz_localize(event_times.index.tz)
     return shift_amount != 0, shift_amount
+
+
+def _has_dst(event_times, date, window, min_difference):
+    before = event_times[date - window:date - pd.Timedelta(days=1)]
+    after = event_times[date:date + window]
+    if len(before) == 0 or len(after) == 0:
+        raise ValueError(f"Insufficient data at {date}.")
+    before = before.dt.hour * 60 + before.dt.minute
+    after = after.dt.hour * 60 + after.dt.minute
+    return abs(before.mean() - after.mean()) > min_difference
+
+
+def has_dst(event_times, shift_dates, window=7, min_difference=45):
+    """Return true if `daynight_mask` appears to have daylight-savings shifts
+    at the dates in `shift_dates`.
+
+    Compares the mean event time in minutes since midnight over the
+    `window` days before and after each date in `shift_dates`. If the
+    difference is greater than `min_difference` then a shift has occurred
+    on that date.
+
+    Parameters
+    ----------
+    event_times : Series
```

Comment:
not wild about `event_times` or `shift_dates`. Maybe `events`? Maybe `times_to_test`?  For `shift_dates` maybe `dst_shift_dates`?